server/routerlicious/packages/services/src/storageNameProvider.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IStorageNameProvider, ITenantManager } from "@fluidframework/server-services-core";
+
+export class StorageNameProvider implements IStorageNameProvider {
+	public constructor(private readonly tenantManager: ITenantManager) {
+	}
+
+	public async getStorageName(tenantId: string, documentId: string): Promise<string> {
+		return this.getTenantStorageName(tenantId);
+	}
+	
+	public async createStorageName(tenantId: string, documentId: string): Promise<string> {
+		return this.getTenantStorageName(tenantId);
+	}
+
+	private async getTenantStorageName(tenantId: string): Promise<string> {
+		const tenant = await this.tenantManager.getTenantConfig(tenantId);
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+		return tenant?.customData.storageName;
```

Comment:
It is because customer data is a key to many dictionary:
![image](https://user-images.githubusercontent.com/3719246/233404863-ffa8ad54-9090-40a2-92d2-335b7e5a77da.png)

I will cast it to string then

Reply:
Ah, but what about when `tenant` is `undefined`? You guard against that with a `?`, which would mean that it _could_ be `undefined`.

Reply:
This file removed